splunklib/binding.py

```diff
@@ -814,13 +814,27 @@ def request(self, path_segment, method="GET", headers=None, body="",
         path = self.authority \
             + self._abspath(path_segment, owner=owner,
                             app=app, sharing=sharing)
+
         all_headers = headers + self.additional_headers + self._auth_headers
         logging.debug("%s request to %s (headers: %s, body: %s)",
                       method, path, str(all_headers), repr(body))
-        response = self.http.request(path,
+
+        if body:
```

Comment:
Let's clean this up so we're only making on call to `self.http.request` and only modifying the dict we pass in